### ISSUE TITLE
Release checklist 2.14 (and 2.14.1)

### ISSUE THREAD

**Issue by devincowan** [hydroshare/hydroshare#5332]
### Issues in Release
- [x] #5228 by @devincowan 
- [x] #5404 by @devincowan 

#### These will be closed as resolved or no longer relevant
- #5418 by @devincowan 
- #5235 by @devincowan 
- #5384 by @devincowan 
- #5195 by @devincowan 
- #5286 by @devincowan 

### Beta Deployment
- [x] Diff RC to master to identify and manually make changes in the following files:
  * hsctl
  * config/hydroshare-config.yaml
  * nginx/config-files/hydroshare-ssl-nginx.conf.template
  * scripts/templates/docker-compose.template
- [x] Changes in hydroshare/local_settings.py need to be coordinated for manual update
- [x] Smoke test active worker before swap, note whether maintenance will be needed (make sure that you check the active worker AFTER the db migrations have run)
- [x] Add any management commands necessary for the deploy to the notes section
- [x] Deployed to Beta
- [x] Review the search and discovery pages
- [x] Create a new user and update profile
- [x] Create iROD account, test connection and delete iROD account
- [x] Create a new resource, check sharing/permission settings, delete new resource
- [x] Developers test around issues
- [x] Hsclient tests pass when targeting beta
- [x] QA testing around issues
- [x] Stakeholders approval

### Production Deployment
- [x] Snapshot the DB and iRods VM before deploy
- [x] Deployed to Production
  * Make manual changes to files identified in Beta Deployment
- [x] Review the search and discovery pages
- [x] Create a new user and update profile
- [x] Create iROD account, test connection and delete iROD account
- [x] Create a new resource, check sharing/permission settings, delete new resource
- [x] Developers test around issue

### Notes relevant to deployment
1. Users over quota need to be contacted
2. Includes migrations, snapshots should be taken
3. Will require ~1hr downtime to run a management command to migrate quota holder info from irods to django `migrate_quota_holder_from_irods`
4. All management commands listed [here](https://github.com/hydroshare/hydroshare/issues/5476#issuecomment-2103008703) must complete before the quotas will be accurate
5. Userzone AVUs should be reset to ensure accuracy
6. Quota enforcement will still be turned off after the deployment. It can be toggled on using the admin panel
7. Quota messages should be checked and updated in the admin panel.
8. Disable the datazone microservices on prod (to avoid the overhead)

### Diff
https://github.com/hydroshare/hydroshare/compare/2.13.10...rc-2.14

**Comment by devincowan** [comment]
beta migration
https://new-hs-ci.hydroshare.org/view/02-Deployment/job/Beta%20deployment/job/beta-hsctl-command/600/console

**Comment by devincowan** [comment]
hsclient tests running here:
https://github.com/hydroshare/hsclient/actions/runs/9021714222

**Comment by devincowan** [comment]
`quota_reset` running here: https://new-hs-ci.hydroshare.org/view/02-Deployment/job/Beta%20deployment/job/beta-hsctl-command/602/console

**Comment by devincowan** [comment]
`dump_user_quota user_quota_050924_beta.csv --exceeded --expand_held_resources`
https://new-hs-ci.hydroshare.org/view/02-Deployment/job/Beta%20deployment/job/beta-hsctl-command/605/parameters/

**Comment by devincowan** [comment]
`reset_file_size_system_metadata --update --refreshed_weeks 52`
https://new-hs-ci.hydroshare.org/view/02-Deployment/job/Beta%20deployment/job/beta-hsctl-command/606/console

**Comment by devincowan** [comment]
I found an issue in the rc 
![image](https://github.com/hydroshare/hydroshare/assets/17934193/9c2675c8-d4f6-4193-8299-916d103e3be0)
likely due to bootstrap upgrade

**Comment by devincowan** [comment]
deploy fails with
```log
REBUILD SOLR INDEX:
  - docker exec hydroshare python manage.py rebuild_index --noinput
System check identified some issues:

WARNINGS:
?: (debug_toolbar.W006) At least one DjangoTemplates TEMPLATES configuration needs to have APP_DIRS set to True.
	HINT: Use APP_DIRS=True for at least one django.template.backends.django.DjangoTemplates backend configuration.
Removing all documents from your index because you said so.
System check identified some issues:

WARNINGS:
?: (debug_toolbar.W006) At least one DjangoTemplates TEMPLATES configuration needs to have APP_DIRS set to True.
	HINT: Use APP_DIRS=True for at least one django.template.backends.django.DjangoTemplates backend configuration.
All documents removed.
System check identified some issues:

WARNINGS:
?: (debug_toolbar.W006) At least one DjangoTemplates TEMPLATES configuration needs to have APP_DIRS set to True.
	HINT: Use APP_DIRS=True for at least one django.template.backends.django.DjangoTemplates backend configuration.
[ERROR/MainProcess] Error updating hs_core using default 
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
psycopg2.errors.UndefinedColumn: column hs_core_genericresource.quota_holder_id does not exist
LINE 1: ...d", "hs_core_genericresource"."spam_allowlisted", "hs_core_g...
```
